File: python/twitter/posts_since.py
# Can I get X (Twitter) posts by date range?
# https://openhandle.dev/questions/can-i-get-x-posts-by-date-range
# Run: python twitter/posts_since.py

import os

from openhandle import OpenHandle

openhandle = OpenHandle(api_key=os.environ["OPENHANDLE_TEST_KEY"])

from datetime import datetime, timezone

# Pagination stops once posts are older than `since`. Filter the end date yourself.
since = datetime(2026, 8, 1, tzinfo=timezone.utc)
until = datetime(2026, 9, 1, tzinfo=timezone.utc)

for post in openhandle.twitter.profile("copperfield_lab_x_test").posts.items(since=since):
    if datetime.fromisoformat(post["createdAt"].replace("Z", "+00:00")) < until:
        print(post["createdAt"], post["text"])

File: typescript/twitter/posts-since.ts
// Can I get X (Twitter) posts by date range?
// https://openhandle.dev/questions/can-i-get-x-posts-by-date-range
// Run: node twitter/posts-since.ts (Node 24 or newer)

import { OpenHandle } from '@openhandle/sdk';

const openhandle = new OpenHandle({ apiKey: process.env.OPENHANDLE_TEST_KEY! });

// Pagination stops once posts are older than `since`. Filter the end date yourself.
const since = '2026-08-01T00:00:00Z';
const until = new Date('2026-09-01T00:00:00Z');

let page = await openhandle.twitter.profile('copperfield_lab_x_test').posts.list({ since });

while (true) {
    for (const post of page.data) {
        if (new Date(post.createdAt) < until) console.log(post.createdAt, post.text);
    }
    const next = await page.next();
    if (!next) break;
    page = next;
}
